### PR TITLE
added blog on registration and authentication in django using loginra…

### PR DESCRIPTION
**Pull request by Tammibriggs**
…dius

#### Please note that in case of below checklist not updated accordingly, the maintainers have the right to immediately close your pull request.

- [x] You created a branch from `dev` branch.
- [x] You PR is raised on `dev` branch and not on`master`.
- [x] Blog follows [Blog Guidelines](https://github.com/LoginRadius/engineering-portal/blob/master/GUIDELINES.md).
- [x] You have read the [Contribution Guidlines](https://github.com/LoginRadius/engineering-portal/blob/master/CONTRIBUTING.md) before creating this PR.
- [x] Checked for any duplicate blog entry.
- [x] Checked for all grammatical errors in the blog.
- [x] You have not published the blog content anywhere else on the internet on platforms like medium, dev.to etc.
- [x] The content is not copy-pasted from anywhere else on the Internet.

##### Author Details for Blog Post Entry

id: Taminoturoko Briggs
bio: An enthusiastic software developer conversant with Python and JavaScript. 
github: https://github.com/Tammibriggs
stackoverflow: https://stackoverflow.com/users/15485047/tammibriggs?tab=topactivity
linkedin: https://www.linkedin.com/in/taminoturoko-briggs-b98251203/

### CODE REVIEW

**Review by abhir9 — APPROVED** [review]
Looks good from technical side

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+Edit the *views.py* file of the **radiusApp** application and add the following code:
+
+*radiusApp/views.py*
+
+    from django.shortcuts import render
+    
+    def register_n_login(request):
+        return render(request, 'index.html')
+
+all this views does is to render a html page as a response.
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:195]:
```suggestion
All this view does is render an HTML page as a response.
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+Now we need to add the URL pattern for this views.
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:196]:
```suggestion
Now, we need to add the URL pattern for this view.
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+*radiusApp/urls.py*
+
+    from django.urls import path
+    from . import views
+    
+    urlpatterns = [
+        path('', views.register_n_login, name="access"),
+    ]
+
+The `register_n_login` views can now be accessed by a URL. It's time to create a template for our views. But before we do that, let me introduce some basic styling we will use in our application.
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:208]:
```suggestion
The `register_n_login` view can now be accessed by a URL. It's time to create a template for our views. But before we do that, let me introduce some basic styling we will use in our application.
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 4]
```diff
@@ -0,0 +1,522 @@
...
+    
+    .profile__details span{
+        font-size: 1.2rem;
+    }
+    
+    .label{
+        color: hsl(210, 7%, 55%);
+    }
+
+Now lets create the template for our `register_n_login` views
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:314]:
```suggestion
Now, let's create the template for our `register_n_login` views
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 5]
```diff
@@ -0,0 +1,522 @@
...
+                    <button onclick="window.location.href='https://{APP_NAME}.hub.loginradius.com/auth.aspx?action=login&return_url=http://127.0.0.1:8000/profile/';">
+                        Login
+                    </button>
+                </div>
+    
+            </div>
+        </body>
+    </html>
+
+We used an `onclick` event on both the register and login button, which when clicked changes the windows location to the [Auth Page (IDX)](https://www.loginradius.com/docs/developer/concepts/idx) register and login page respectively.
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:361]:
```suggestion
We used an `onclick` event on both the register and login buttons, which when clicked changes the window's location to the [Auth Page (IDX)](https://www.loginradius.com/docs/developer/concepts/idx) register and login page respectively.
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 6]
```diff
@@ -0,0 +1,522 @@
...
+
+
+> NOTE: Don’t forget to replace the {APP_NAME} placeholder with your LoginRadius app name.
+>
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:365]:
```suggestion
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 7]
```diff
@@ -0,0 +1,522 @@
...
+        </body>
+    </html>
+
+We used an `onclick` event on both the register and login button, which when clicked changes the windows location to the [Auth Page (IDX)](https://www.loginradius.com/docs/developer/concepts/idx) register and login page respectively.
+
+
+> NOTE: Don’t forget to replace the {APP_NAME} placeholder with your LoginRadius app name.
+> 
+
+Also, in the register and login link, we replaced the {RETURN_URL} parameter with http://127.0.0.1:8000/profile/, which is the users profile page we are going to create shortly.
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:367]:
```suggestion
Also, in the register and login link, we replaced the {RETURN_URL} parameter with http://127.0.0.1:8000/profile/, which is the user profile page we are going to create shortly.
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 8]
```diff
@@ -0,0 +1,522 @@
...
+
+Run the development server and open http://127.0.0.1:8000/, you will see a page like this:
+
+![Register and Login page](https://paper-attachments.dropbox.com/s_06CCF8584138C77AD580AFE76E986CC1E8C4C360158F4A6548D0F9EAC304864E_1626849064162_Screenshot+48.png)
+
+
+Now when you click the register or login button your will see a page like this:
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:374]:
```suggestion
Now, when you click the register or login button, you will see a page like this:
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 9]
```diff
@@ -0,0 +1,522 @@
...
+
+![Auth page (IDX)](https://paper-attachments.dropbox.com/s_06CCF8584138C77AD580AFE76E986CC1E8C4C360158F4A6548D0F9EAC304864E_1626849029873_Screenshot+46.png)
+
+> When LoginRadius successfully authenticates a user, it attaches an access token in the query string as a token parameter to the REDIRECT_URL.
+
+we are going to retrieve the user's data from the access token.
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:380]:
```suggestion
We are going to retrieve the user's data from the access token.
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 10]
```diff
@@ -0,0 +1,522 @@
...
+
+    from LoginRadius import LoginRadius as LR
+    
+    LR.API_KEY = "API_KEY"
+    LR.API_SECRET = "API_SECRET"
+    loginradius = LR()
+
+Replace API_KEY and API_SECRET with your respective LoginRadius API credentials.
+
+Now lets create the view that will be responsible for retrieving user data
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:404]:
```suggestion
Now, let's create the view that will be responsible for retrieving user data.
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 11]
```diff
@@ -0,0 +1,522 @@
...
+        else:
+            return redirect('access')
+    
+        return render(request, 'profile.html', {'user_data':user_data})
+
+In this code snippet, we imported the initialized LoginRadius class from the *config.py* file and also imported `redirect` and `HttpResponse` from there respective modules.
+We also introduced a new function called `profile`. In this function we did the following:
+
+- We first checked if there is a token parameter attached to the request. If true,
+- We created a `try` and `except` block. In the `try` block we used the `authentication.get_profile_by_access_token` method to retrieve the user data from the access token and assigned it to a variable called `user_data` while in the `except` block we returned a HttpResponse of the error that occurred if the `try` block fails.
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:430]:
```suggestion
- We first checked if there is a token parameter attached to the request.
- If true, we created a `try` and `except` block. In the `try` block, we used the `authentication.get_profile_by_access_token` method to retrieve the user data from the access token and assigned it to a variable called `user_data`; while in the `except` block, we returned a HttpResponse of the error that occurred if the `try` block fails.
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 12]
```diff
@@ -0,0 +1,522 @@
...
+- If the `try` block successfully retrieved the user's data, we rendered a *profile.html* page, and also attached the `user_data` variable to the response.
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:431]:
```suggestion
- If the `try` block successfully retrieved the user's data, we rendered a *profile.html* page and also attached the `user_data` variable to the response.
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 13]
```diff
@@ -0,0 +1,522 @@
...
+                        <span class="label">Number of login:</span> {{user_data.NoOfLogins}}
+                        </span>
+                    </div>
+                </div>            
+            </div>
+    
+        </body>
+    </html>
+
+Here we displayed the Provider, Email and NoOfLogins from the `user_data`. These are just some of the data present in the `user_data`.
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:487]:
```suggestion
Here, we displayed the Provider, Email, and NoOfLogins from the `user_data`. These are just some of the data points present in the `user_data`.
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 14]
```diff
@@ -0,0 +1,522 @@
...
+Now if you run your development server then register and login a user  you should see a page like this:
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:488]:
```suggestion
Now, if you run your development server, and then register and login a user, you should see a page like this:
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 15]
```diff
@@ -0,0 +1,522 @@
...
+    
+            {{user_data}}
+    
+        </body>
+    </html>
+
+
+## Conclusion
+
+In this tutorial we have looked at how we can use LoginRadius for registration and authentication in Django. We started by giving a brief introduction to LoginRadius, what it is, its benefits, and how we can get started with it, then we rounded up by building a demo application with Django were implemented registration and authentication using LoginRadius.
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:516]:
```suggestion
In this tutorial, we discussed how to use LoginRadius for registration and authentication in Django. We started by introducing LoginRadius, what it is, its benefits, and how to get started with it. Then, we illustrated building a demo application with Django and implemented registration and authentication using LoginRadius.
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 16]
```diff
@@ -0,0 +1,522 @@
...
+
+
+## Github repo
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:519]:
```suggestion
**GitHub repo:** https://github.com/Tammibriggs/radiusAuth
```

File: content/registration-and-authentication-in-django-using-loginRadius/index.md [hunk 17]
```diff
@@ -0,0 +1,522 @@
...
+https://github.com/Tammibriggs/radiusAuth
```
Comment (raghunath-r-a) [inline content/registration-and-authentication-in-django-using-loginRadius/index.md:520]:
```suggestion
```

**Review by raghunath-r-a — CHANGES_REQUESTED** [review]
Hi @Tammibriggs 

We have made suggestions to improve the article. Please review them and make the necessary changes. 

Thank you,
@raghunath-r-a

**Comment by Tammibriggs** [comment]
Hi,

I have made the suggested changes.

Regards,
Taminoturoko Briggs.

**Review by raghunath-r-a — APPROVED** [review]
Thank you, @Tammibriggs

**Review by abhir9 — APPROVED** [review]

**Pull request merged**